### DISCUSSION TITLE
How to ignore null datasets for tooltip.

### DISCUSSION THREAD

**Discussion by master117** [recharts/recharts#4423] [General]
Hello. I have weekly data which I'm showing in a line chart. Additionally I'm using different colored reference areas to mark the months of the year.

As we all know week starts to not perfectly overlap with month starts. I can only create reference areas on data points. So I'm adding empty data points, where I have a date but no values.

But when hovering over the graph to see my values in tool tips it will not snap to the actual data points. I assume it's stuck on the dummy values, even if it doesn't display anything.

I'm not sure if this is a bug as it's homing in on things it's not actually displaying, but either way it's unsuitable for me.

Cheers

**Comment by ckifer** [comment]
Can you create a code sandbox example of this?

[![Edit simple-line-chart (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-line-chart-forked-gwll7p)

That is as far as I can get without more information. How exactly do you have things set up?

I can set `cursor={false}` if you don't mind losing the cursor for all data points, but otherwise it will track to the data index that it is currently on (null and all)

**Comment by master117** [comment]
Thanks already @ckifer 

I have adjusted it to fit more closely to how it looks for me:

https://codesandbox.io/p/sandbox/simple-line-chart-forked-xhsrtm

You will notice that you have to get super close to the points to be able to show them, even if all the points around it are null.

**Reply by ckifer** [reply]
you could do something like

`<Tooltip cursor={<></>} />` which removes the cursor but will still render the Tooltip if you are on that index.

But there aren't currently any ways to increase the "tolerance" or the trigger area that will make the Tooltip appear

----

You can add a second X Axis that represents month of the year only, differentiate by xAxisId and then `hide` it

kind of like this
```
      <XAxis dataKey="name" xAxisId="test" />
      <XAxis dataKey="month" xAxisId="test2" hide />
      <YAxis />
      <Tooltip cursor={<></>} />
      <Legend />
      <Line
        type="monotone"
        dataKey="pv"
        stroke="#8884d8"
        activeDot={{ r: 8 }}
        connectNulls
        xAxisId="test"
      />
      <ReferenceArea
        x1={"PageA"}
        x2={"PageF6"}
        fill="red"
        ifOverflow="visible"
        xAxisId="test2"
      />
```

but I'm not sure that would accomplish what you want

**Reply by master117** [reply]
That idea with a second xAxis is genius. I never realized you could do something like that. Will try it out later. Thank you so much my dude.

**Reply by master117** [reply]
Ok, I just tried your idea and sadly it does not work.

https://codesandbox.io/p/sandbox/simple-line-chart-forked-xhsrtm

![grafik](https://github.com/recharts/recharts/assets/444009/6accd4d3-e396-4e3b-a91a-598d69cd69ae)

It still creates empty ticks/gridlines for null or undefined fields. Which I find very strange behavior. And these also still eat the cursor.

Any other ideas?

**Reply by ckifer** [reply]
Last resort is custom work using `Customized />` component

[![Edit custom-reference-areas](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/simple-line-chart-forked-j5jyjw?file=%2Fsrc%2FApp.tsx%3A79%2C11)

Unfortunately I think this ends in the same result as above though (and is way more complicated).

There are some things you can do to get rid of the grid lines such as `verticalValues`

For tick lines you'd have to be able to programmatically disable individual tick lines which you can't right now unfortunately
